Migrating TravisCI to xenial and beyond

PhilMFischer opened the pull request:
- Changed travis script to got to xenial
- Added new apt dependencies for loading zmq and vtk 
- Refactored external lib plugins for linux now using external libraries rather then shiopped ones
- added new script to prepare the environment
- Still with openJDK 8, therefore not yet on  bionic which does not support 8 anymore.
- Removed all so files which were sled dependant

:warning: Be careful with merging, all other depending projects have to be prepared, otherwise they will fail to build on TravisCI due to missing environment. :warning:

Commit:
Migrated Travis CI config - (Task #201)

- removed distribution fix to trusty, possible the build will be
executed is bionical
- fixed java from oraclejdk8 to openjdk8
- set xvfb as service rather than starting it up manually

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Working on linux library loading - (Task #21)

- Worked on both zmq and vtk
- Now reading libraries from external environment variable
- Removed shipped libraries from plugins
- Changed build.properties to not include the jar in the binary build
anymore. But the ajr is still present and linked in the classpath to
make the code compile in eclipse.
- Did a minor fix to the NativeLibraryLoader when not Native Library is
specified.

---
Task #21: Update dependencies (Eclipse, XText, etc)

Commit:
Adding new script to setup build environment

Commit:
Removing warnings on build properties - (Task #201)

- Build properties complaining about reference to external jar
- Adjusted project settings to ignore warning.

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Merge branch 'feature/task_21-Update_Linux' into feature/task_201-Migrate_TravisCI_buil

Commit:
Merge remote-tracking branch 'origin/development' into feature/task_201-Migrate_TravisCI_buil

Commit:
Fix warning on excel plugin - (Task #201)

- Set PDE compiler to ignore other warnings on manifest and build
problems.

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Trying to build with xenial - (Task #201)

- Added vtk and zmq to apt packages
- Switched to xenial distribution
- Switched to earlier versions of vtk and zmq in scripts due to xenial

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Ignore incompatible environment warning - (Task #201)

- Windows is first plattform complaints in linux plugins are not that
concerning right now.

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Removed warning to get travis running again - (Task #201)

- Removed warning from NativeLib plugin, due to changes in how we deal
with linux plugins. Needs major refactoring.

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Removing vtk-jni from apt - (Task #201)

- for vtk 6 there is no vtk-jni apckage

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Fixing typo in travis script - (Task #201)

- Corrected spelling of setup_environment script

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Double typo fixed - (Task #201)

- the script was not correctly named as well

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Added new class to SWTBOT Test for tracing - (Task #201)

- Implemented new Rule that tells when a test gets called and when it is
left. This should help to understand where SWTBot gets stuck during
TravisCI builds

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Modifying activator to read from vtkNativelibraries - (Task #201)

- Readjusting Plugin Library loading

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Merge branch 'feature/task_201-Migrate_TravisCI_buil' of https://github.com/virtualsatellite/VirtualSatellite4-Core.git into feature/task_201-Migrate_TravisCI_buil

Commit:
Refactored Library Loaders - (Task #201)

- reactivated Activators for loading libraries into plugins fior zmq and
vtk
- Removed Native VTK lib loading from visualization ui plugin.

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Try to find vtk jar - (Task #201)

- Build process does not yet find vtk jar

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Chasing issue with linking to vtk - (Task #201)

- External vtk lib is not linked into project.

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Chasing issue with linking to vtk - (Task #201)

- trying to identify the vtk.jar

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Chasing issue with linking to vtk - (Task #201)

- Found vtk6.jar. Check if if variable is visible to build itself.

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Chasing issue with linking to vtk - (Task #201)


---
Task #201: Migrate TravisCI build nodes to xenial or beyond

Commit:
Adding local jar to classpath - (Task #201)

- Adding local jar to class path but not deploying them in build jar


---
Task #201: Migrate TravisCI build nodes to xenial or beyond

SaMuellerDLR commented on de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java:
```diff
@@ -52,6 +55,19 @@
+	public TestRule testStatusWatcher = new TestWatcher() {
+		@Override
+		protected void starting(Description description) {
+			System.err.println("SWTBotTestCase: " + description.getMethodName() + "...started");
```
Any reason to output it in System.err instead of System.out?

PhilMFischer replied:
The basic idea was to see if it is possible to get direct out put into the TravisCI log rather then the redirected EclipseLog which is redirected into the file. Since it did not work i changed it to standard out now.

Commit:
Printing test case names to System out. - (Task #201)

- Tried error to see if it is possible to directly output into the
TraviCI log.

---
Task #201: Migrate TravisCI build nodes to xenial or beyond

SaMuellerDLR reviewed (approved):
Approved
